sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -93,29 +93,35 @@ def create_experiment(
 
         namespace = namespace or self.namespace
 
+        if 'name' in experiment.metadata and experiment.metadata.name:
```

Comment:
I think, it will raise Exception if `experiment.metadata.name` is not set.
We might need to have something like this:
```python
         experiment_name = None

        if type(experiment) == models.V1beta1Experiment:
            if experiment.metadata.name is not None:
                experiment_name = experiment.metadata.name
            elif experiment.metadata.generate_name is not None:
                experiment_name = experiment.metadata.generate_name
        elif "name" in experiment["metadata"]:
            experiment_name = experiment["metadata"]["name"]
        elif "generateName" in experiment["metadata"]:
            experiment_name = experiment["metadata"]["generateName"]
        if experiment_name is None:
            raise ValueError("Experiment must have name or generateName")
```
Any thoughts @droctothorpe @tenzen-y @johnugeorge @bharathk005 @fensterwetter

Reply:
@andreyvelich  thanks for the review.
I have added some changes based on your suggestions. Test cases pass on my machine.

Regarding the type assertion, the function already expects a "models.V1beta1Experiment" type:

https://github.com/kubeflow/katib/blob/1039f4b1bd37ee7b2e0e5e94394d25dfdbd3dd05/sdk/python/v1beta1/kubeflow/katib/api/katib_client.py#L80

Please let me know if we still need to check for the type.

Reply:
> I think, it will raise Exception if `experiment.metadata.name` is not set. We might need to have something like this:
> 
> ```python
>          experiment_name = None
> 
>         if type(experiment) == models.V1beta1Experiment:
>             if experiment.metadata.name is not None:
>                 experiment_name = experiment.metadata.name
>             elif experiment.metadata.generate_name is not None:
>                 experiment_name = experiment.metadata.generate_name
>         elif "name" in experiment["metadata"]:
>             experiment_name = experiment["metadata"]["name"]
>         elif "generateName" in experiment["metadata"]:
>             experiment_name = experiment["metadata"]["generateName"]
>         if experiment_name is None:
>             raise ValueError("Experiment must have name or generateName")
> ```
> 
> Any thoughts @droctothorpe @tenzen-y @johnugeorge @bharathk005 @fensterwetter

Additionally, I would suggest to add the Exception if both `name` and `generateName` are specified.
@andreyvelich WDYT?

Reply:
Won't Kubernetes Python client throw an error when we try to create custom resources in that case ?

Reply:
>Regarding the type assertion, the function already expects a "models.V1beta1Experiment" type:

That's true, but since Python is not strong typed language, users can still send `dict` as Experiment. Otherwise, we need to check `type(experiment)` and through exception. Do we want to give users functionality to pass dict with `create_experiment` API @droctothorpe @tenzen-y @johnugeorge ?

Reply:
> Won't Kubernetes Python client throw an error when we try to create custom resources in that case ?

I imaged such like 102-103:

https://github.com/kubeflow/katib/blob/5be7150ac760c800bdab156a01dff290ec67d3bd/sdk/python/v1beta1/kubeflow/katib/api/katib_client.py#L102-L103

So, my concerns have gone away. Thanks.

Reply:
> That's true, but since Python is not strong typed language, users can still send dict as Experiment. Otherwise, we need to check type(experiment) and through exception. Do we want to give users functionality to pass dict with create_experiment API @droctothorpe @tenzen-y @johnugeorge ?

I agree with the necessary for type assertion, but I think this is PR to support name generation.
So, I'm ok with working on it in another PR.

Reply:
@tenzen-y @andreyvelich thanks for the comments.
1. Type assertion
2. Throwing error if both name and generateName are provided
Are these the changes that need to be implemented?

Reply:
@tenzen-y @andreyvelich Thanks for your time. I have pushed a commit with the changes mentioned in this thread. Please review and let me know.